CONTRIBUTING.md

```diff
@@ -1,39 +1,145 @@
-Contributing to GraSPy
-======================
+# Contributing
```

Comment:
I'm going to add more onto this later, as well as indicate that we're going to have a pre-merge check that actually does the following for you:
- mypy check
- black formatting check (specifically looking to make it exit with error if it had to change anything, which will fail the build)
- pytest check
- sphinx doc generation with doctest verification (failure upon any error or warning)

The PR won't even be in a mergeable state until all of these succeed.  We'll tell the user about this process but also tell them how to avoid a round trip by running these on their own system first.